process/block/baseProcess.go

```diff
@@ -417,12 +414,15 @@ func (bp *baseProcessor) requestHeadersIfMissing(sortedHdrs []data.HeaderHandler
 				missingNonces = append(missingNonces, j)
 			}
 		}
+
+		highestHdr = currHdr
 	}
 
 	// ask for headers, if there most probably should be
-	if len(missingNonces) == 0 && !isLastNotarizedCloseToOurRound {
-		startNonce := prevHdr.GetNonce() + 1
-		for nonce := startNonce; nonce < startNonce+process.MaxHeaderRequestsAllowed; nonce++ {
+	if maxRound > highestHdr.GetRound() {
```

Comment:
this if is always true. I do not agree with the changes did in this function. Now missingNonces may have duplicates and you will ask for duplicates. keep only unique elements in missingNonces.